### PR TITLE
Physics

### PR DESCRIPTION
**Pull request by jabenk**
Merge current physics state.

### CODE REVIEW

File: packages/sdk/src/core/contextInternal.ts [hunk 1]
```diff
@@ -114,23 +135,24 @@ export class ContextInternal {
...
 		let user: User;
-		if (actor.exclusiveToUser &&
-			(user = this.userSet.get(actor.exclusiveToUser)) &&
-			!(user.grantedPermissions.includes(Permissions.UserInteraction))) {
-			actor.internal.notifyCreated(false,
-				`Permission denied on user ${user.id} (${user.name}). Either this MRE did not ` +
-				"request the UserInteraction permission, or it was denied by the user."
-			);
+		if (actor.exclusiveToUser) {
+			user = this.userSet.get(actor.exclusiveToUser);
```
Comment (jabenk) [inline packages/sdk/src/core/contextInternal.ts:139]: TODO: this should be fixed before merge

File: packages/sdk/src/core/contextInternal.ts [hunk 2]
```diff
@@ -114,23 +135,24 @@ export class ContextInternal {
...
-		if (actor.attachment?.userId &&
-			(user = this.userSet.get(actor.attachment?.userId)) &&
-			!(user.grantedPermissions.includes(Permissions.UserInteraction))) {
-			actor.internal.notifyCreated(false,
-				`Permission denied on user ${user.id} (${user.name}). Either this MRE did not ` +
-				"request the UserInteraction permission, or it was denied by the user."
-			);
+		if (actor.attachment?.userId) {
+			user = this.userSet.get(actor.attachment?.userId);
+			//if (!(user.grantedPermissions.includes(Permissions.UserInteraction))) {
```
Comment (jabenk) [inline packages/sdk/src/core/contextInternal.ts:151]: TODO: this should be fixed before merge

File: packages/functional-tests/src/tests/physics-pile-test.ts [hunk 3]
```diff
@@ -13,6 +13,8 @@ import { User } from '@microsoft/mixed-reality-extension-sdk';
 
 export default class PhysicsPileTest extends Test {
 
+	public expectedResultDescription = "Piles should apear similar for each client";
```
Comment (tombuMS) [inline packages/functional-tests/src/tests/physics-pile-test.ts:16]: >apear [](start = 50, length = 5)

typo

File: packages/functional-tests/src/tests/physics-stack-test.ts [hunk 4]
```diff
@@ -7,12 +7,12 @@ import * as MRE from '@microsoft/mixed-reality-extension-sdk';
 
 import { App } from '../app';
 import { Test } from '../test';
-import { Vector3 } from '@microsoft/mixed-reality-extension-sdk';
+import { Vector3, Actor } from '@microsoft/mixed-reality-extension-sdk';
```
Comment (tombuMS) [inline packages/functional-tests/src/tests/physics-stack-test.ts:10]: These are already imported through the above MRE import.  Use MRE.Actor and MRE.Vector3 and remove this added import.

File: packages/sdk/src/internal/subscriptionType.ts [hunk 5]
```diff
@@ -7,4 +7,4 @@
  * A type representing the different parts of an Actor that can be subscribed to. When subscribed, the Actor will
  * receive updates when corresponding changes occur on the host.
  */
-export type SubscriptionType = 'transform' | 'rigidbody' | 'collider';
+export type SubscriptionType = 'transform' | 'rigidbody' | 'collider' | 'rigidbodyVelocity';
```
Comment (tombuMS) [inline packages/sdk/src/internal/subscriptionType.ts:10]: >rigidbodyVelocity [](start = 73, length = 17)

Naming conventions for our string types are hyphen separated.  Should be `rigidbody-velocity I believe,

File: packages/sdk/src/actor/physics/physicsBridge.ts [hunk 6]
```diff
@@ -4,10 +4,24 @@
...
 
 export type PhysicsBridgeTransformUpdate = {
     id: Guid;
     time: number;
     transformCount: number;
+    flags: number;
     transformsBlob: string;
 }
+
+export type PhysicsServerOneTransformsUpdate = {
```
Comment (tombuMS) [inline packages/sdk/src/actor/physics/physicsBridge.ts:17]: >PhysicsServerOneTransformsUpdate [](start = 12, length = 32)

This feels like a slightly awkward name.  Would `PhysicsServerTransformUpdate` be better and more concise?

File: packages/sdk/src/core/contextInternal.ts (hunk 1 above)
Comment (tombuMS) [reply packages/sdk/src/core/contextInternal.ts:139]: Is the code getting uncommented as a part of the merge?

---
In reply to: [449561360](https://github.com/microsoft/mixed-reality-extension-sdk/pull/652#discussion_r449561360) [](ancestors = 449561360)

File: packages/sdk/src/core/contextInternal.ts (hunk 2 above)
Comment (tombuMS) [reply packages/sdk/src/core/contextInternal.ts:151]: Is the code getting uncommented as a part of preparing this for a merge?

---
In reply to: [449561436](https://github.com/microsoft/mixed-reality-extension-sdk/pull/652#discussion_r449561436) [](ancestors = 449561436)

File: packages/sdk/src/actor/physics/physicsBridge.ts (hunk 6 above)
Comment (jabenk) [reply packages/sdk/src/actor/physics/physicsBridge.ts]: Changed from PhysicsServerOneTransformsUpdate to PhysicsServerOneTransformUpdate

---
In reply to: [450331004](https://github.com/microsoft/mixed-reality-extension-sdk/pull/652#discussion_r450331004) [](ancestors = 450331004)

File: packages/sdk/src/core/contextInternal.ts (hunk 1 above)
Comment (jabenk) [reply packages/sdk/src/core/contextInternal.ts:139]: Added user.hasGrantedPermissions

---
In reply to: [450333599](https://github.com/microsoft/mixed-reality-extension-sdk/pull/652#discussion_r450333599) [](ancestors = 450333599,449561360)

File: packages/sdk/src/core/contextInternal.ts (hunk 2 above)
Comment (jabenk) [reply packages/sdk/src/core/contextInternal.ts]: Added user.hasGrantedPermissions

---
In reply to: [450333867](https://github.com/microsoft/mixed-reality-extension-sdk/pull/652#discussion_r450333867) [](ancestors = 450333867,449561436)

**Pull request merged**